### ISSUE TITLE
[CLI]: Wandb config and summary not fully saved during some offline runs

### ISSUE THREAD

**Issue by CedricLeon** [wandb/wandb#7227]
## Description of the issue
When running sweeps offline, some runs don't get their config and their summary uploaded (in fact not even saved locally).
There is no crash from wandb, nor any error. Some runs config are correctly written (i.e., I can see the config that I gave to `wandb.init()`) and some only contains the `wandb_version` and a `_wandb` object. This is similar for the summary (either it contains all sort of info: train, valid and test metrics, `timestamp`, etc. or only the `runtime`).

## Setup 
I use Hydra and Lightning with W&B, based on this amazing [template](https://github.com/ashleve/lightning-hydra-template) from ashleve. On top of that, I use the submitit launcher plugin of Hydra to generate my SLURM scripts.
As far as I've seen, the problem comes from running offline runs in parallel, but it seems very tricky to narrow down the problem to a minimalist code example I could paste there. I attached 2 zip files containing 2 example runs (one "good": with a full config and all my metrics in the summary, and one "bad") output logs. If you need more information, I would be happy to provide you with more snippets of my code.

Here is how I _manually_ initialize my wandb run:
```python
wandb.config = omegaconf.OmegaConf.to_container(
        cfg, resolve=True, throw_on_missing=True
)
 wandb.init(
        entity=cfg.logger.wandb.entity,
        project=cfg.logger.wandb.project,
        dir=cfg.logger.wandb.save_dir,
        tags=cfg.tags,
        mode="offline" if cfg.logger.wandb.offline else "online",
        settings=wandb.Settings(start_method="thread"),
 )
```
I wrote that code based on this nice [tutorial](https://wandb.ai/adrishd/hydra-example/reports/Configuring-W-B-Projects-with-Hydra--VmlldzoxNTA2MzQw).

If you wish I can also share how i instantiate my Lightning WanbLogger and the Lightning Trainer, but I don't think I do anything fancy. These are all my interactions with wandb (except of course my logs in the LightningModule, which are pretty basic).

## Miscellaneous information
You can check the full config in `.hydra`. The only difference between the runs is the `seed`, the 2 example runs were part of a sweep of 36 runs (the "good" and the "bad" run are number 8 and 9, respectively).

To avoid you the pain of looking for differences between the 2 runs in all the files I attached, here is what I found:
- the main difference lies in `output.log` where the "bad" run logs stop before the testing, as if it crashed (but you can check the SLURM logs, it didn't)
- Otherwise, `wandb/files/config.yaml` and `wandb/files/wandb-summary.json` are of course different

Also, if you look carefully you will see that I use the Lightning Callback called `wandb-osh` from [wandb-offline-sync-hook](https://github.com/klieret/wandb-offline-sync-hook)  in order to live synchronize my runs when running offline. But I already made sure, the problem does not come from here, i.e., running fully offline and performing a `wandb sync ...` once the run is done does not change the problem.

### Additional Files

You can find here 2 zip files containing the output files in both scenari:
- [missing_confi-and-summary_run.zip](https://github.com/wandb/wandb/files/14759410/missing_confi-and-summary_run.zip)
- [correctly_saved-synchronized_run.zip](https://github.com/wandb/wandb/files/14759411/correctly_saved-synchronized_run.zip)

Because of the GitHub upload file size limit I deleted the checkpoints, as well as the `latest_run` folder in `wandb` because it seems to me this is only a copy of the `offline-run...` folder.
Also, I added the `.out` and `.err` output files from the SLURM jobs, just so you can check the full output of the run. 

### Environment

**WandB** version: 0.16.4

**OS**:  Linux-4.18.0-513.18.1.el8_9.x86_64-x86_64-with-glibc2.28

**Python** version: 3.11.7

Versions of relevant libraries (from `conda env list`):
- `torch`: 2.2.0+cu118
- `pytorch-lightning`: 2.2.0.post0
- `hydra-core`: 1.3.2
- `hydra-submitit-launcher`: 1.2.0
- `wandb-osh`: 1.2.2

### Additional Context

I found a way of "fixing" the "bad" runs config. i wrote a script scanning my wandb project for empty config and then overrriding them with the corresponding Hydra config i find in my local logs.
However, doing that for the summary seems way more tricky.

**Comment by CedricLeon** [comment]
Hi @luisbergua, sorry I wasn't super clear, I am using Hydra sweeps, not W&B sweeps. I plan to switch to W&B sweeps, but i haven't figured out a clean way of setting that up with Hydra configs system.
So, I don't see a clean way of sharing you my sweep. I wanted to share the 2 example runs, but from what I read it is not possible to share runs from a private project, only reports, am I right?
I think this also answer the question of the problem happening without Hydra and Lightning.

**Comment by luisbergua** [comment]
Hey @CedricLeon, apologies about the delay and thanks for sharing this information! Would you mind then sharing a link to any of the affected runs? I'd like to take a look at it

**Comment by CedricLeon** [comment]
Hi, no problem I'm also not very active.
I still don't think you can access the runs, these are on my private project 😅, but here are the links:
- [Here](https://wandb.ai/cedric-leonard/CompressAILit/runs/xl9oaqj0/) is the link to an affected run (_xl9oaqj0_).
- [Here](https://wandb.ai/cedric-leonard/CompressAILit/runs/yaxoa2oe/) is the link to an **un**affected run (_yaxoa2oe_).
Both runs correspond to the ones I shared the logs in the zip files.

As an additional information, I upgraded the script I use to update the config of the affected runs to now also partially recover the summary. Beforehand, I could easily see the affected runs because when looking at the overview of a specific run, the config displayed was empty (even though I updated it using my script). However, I can now see the config of the updated runs. I don't know if anything changed meanwhile.

**Comment by luisbergua** [comment]
Hey @CedricLeon, thanks for your answer! I also see the config of your runs so it might have been fixed. Mind confirming if you are able to reproduce the issue in any way?

**Comment by CedricLeon** [comment]
Thanks for checking. The issue is that I ran my script to fix all the affected runs, so I have none left to analyze 😅
I will comment back here as soon as I get a new affected run, but seeing the size of the sweeps I had to run to see the effect, this might take a bit

**Comment by Mjvolk3** [comment]
Hi there. I have an update on this with some minimal scripts to help reproduce the issue.

Using a conda env.


**Wand Version:** 0.17.7

**Chip**: Apple M1

**OS:** Sonoma 14.6.1

**Python Version:** 3.11.7

Here is the minimal script. It loops through some steps and logs data. The last of which we would expect to end up in the `wandb-summary.json`.

```python
#wandb_offline_debug.py
import wandb
import random
import json
import os
import time


def run_experiment(mode):
    # Initialize wandb
    run = wandb.init(project="wandb-offline-debug", mode=mode)

    # Log some random data
    for step in range(100):
        data = {f"metric_{i}": random.random() for i in range(5)}
        wandb.log(data)

    # Print the summary
    print(f"\nMode: {mode}")
    print("Summary:")
    summary_dict = wandb.run.summary._as_dict()
    print(json.dumps(summary_dict, indent=2))

    wandb.finish()
    return run.dir


def print_summary_contents(run_dir):
    summary_file_path = os.path.join(run_dir, "wandb-summary.json")
    if os.path.exists(summary_file_path):
        print(f"\nContents of {summary_file_path}:")
        with open(summary_file_path, "r") as f:
            print(f.read())
    else:
        print(f"\nSummary file not found at {summary_file_path}")


if __name__ == "__main__":
    # Run in online mode
    online_run_dir = run_experiment("online")

    # Run in offline mode
    offline_run_dir = run_experiment("offline")

    # Print contents of both summary files after runs finish
    print("\n--- Final Summary Contents ---")
    print_summary_contents(online_run_dir)
    print_summary_contents(offline_run_dir)

```
From the results we see that std out shows the logged metrics and the summary in the `Run Summary`. Yet when we look at the `wandb-summary.json` these data only appears for the `online` run. When you go to sync the runs obviously the summary metrics won't appear for the offline runs. Just look under `--- Final Summary Contents ---`, the keys of json should match.

```bash
michaelvolk@M1-MV torchcell % /Users/michaelvolk/opt/miniconda3/envs/torchcell/bin/python /Users/michaelvolk/Documents/projects/torchce
ll/wandb_offline_debug.py
wandb: Currently logged in as: mjvolk3 (zhao-group). Use `wandb login --relogin` to force relogin
wandb: Tracking run with wandb version 0.17.7
wandb: Run data is saved locally in /Users/michaelvolk/Documents/projects/torchcell/wandb/run-20240828_164817-fktu3sxg
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run stellar-puddle-5
wandb: ⭐️ View project at https://wandb.ai/zhao-group/wandb-offline-debug
wandb: 🚀 View run at https://wandb.ai/zhao-group/wandb-offline-debug/runs/fktu3sxg

Mode: online
Summary:
{
  "metric_0": 0.715624369777277,
  "metric_1": 0.2355671276911444,
  "metric_2": 0.6348888401726657,
  "metric_3": 0.7421840352522208,
  "metric_4": 0.42073917912566927,
  "_timestamp": 1724881704.603507,
  "_runtime": 7.228076934814453,
  "_step": 99
}
wandb:                                                                                
wandb: 
wandb: Run history:
wandb: metric_0 ▅▄▂▇▆▄▅▃▄█▅▁▁▂▅▄██▇▅▇▆▄▂▃▆▄▃▁▁▂▄▄▆▇▆▁▃▆▆
wandb: metric_1 ▇▂▃▅▂▆█▆▇▂▃▂▅▇█▇▃▅▅▃▃▇▆▃▆▃▇█▆▃▅▁█▂▇▃▄▅▇▂
wandb: metric_2 █▁▃▄▁▄▄▆█▅▅▃▃▅▆▁▃▂▂▄▂█▃▇▆▄▃▇█▇▂▆▇▁█▃█▇▇▅
wandb: metric_3 ▂▄▃▄▁▆▆▁▆▆▆▅▆▄█▆▇▄▅█▇█▃▃▆█▇▁▃▄▂▄█▄▃▁▇▆▁▆
wandb: metric_4 ▄▄▁▁▁▇▆█▆▃▄▄▂▄▂▇▇▄█▃▄▄▂▂▃▇▃▅▄▄▂▅▆▂▂▆█▆▄▄
wandb: 
wandb: Run summary:
wandb: metric_0 0.71562
wandb: metric_1 0.23557
wandb: metric_2 0.63489
wandb: metric_3 0.74218
wandb: metric_4 0.42074
wandb: 
wandb: 🚀 View run stellar-puddle-5 at: https://wandb.ai/zhao-group/wandb-offline-debug/runs/fktu3sxg
wandb: ⭐️ View project at: https://wandb.ai/zhao-group/wandb-offline-debug
wandb: Synced 6 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
wandb: Find logs at: ./wandb/run-20240828_164817-fktu3sxg/logs
wandb: WARNING The new W&B backend becomes opt-out in version 0.18.0; try it out with `wandb.require("core")`! See https://wandb.me/wandb-core for more information.
wandb: Tracking run with wandb version 0.17.7
wandb: W&B syncing is set to `offline` in this directory.  
wandb: Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.

Mode: offline
Summary:
{
  "metric_0": 0.5310924540580804,
  "metric_1": 0.008354595586691382,
  "metric_2": 0.6474718406912846,
  "metric_3": 0.7654659672380345,
  "metric_4": 0.645856304637467,
  "_timestamp": 1724881716.758178,
  "_runtime": 7.470421075820923,
  "_step": 99
}
wandb:                                                                                
wandb: 
wandb: Run history:
wandb: metric_0 ▂▂▂▆▁▂▂▂▃▃▄▆▄▁▄▂▃▆▃▆▆▇▃▃▇▇▃▃▅▆▁▃▃▇█▅▄▃▃▅
wandb: metric_1 ▆▂▅▆▂▃▃▃▇▄█▁▂▁▇▃▄▂▆▃▃▇▄▄▆▇▂██▃▃▃▄█▆▇▆▄▁▁
wandb: metric_2 ▇▇▅▂▃▄█▄▅█▆█▄▂█▇▄▇▃▂▄▄▆▂▃▂▁▄▃▄▇▆▁▆▇▆▅▆█▆
wandb: metric_3 ▂▇▂▅▂▄▂▆▁▇▃▇▃▇▁▂▄▂█▃▂▂▂▁▃▆▆▃▃▁█▅▆▁▄█▇▃▄▆
wandb: metric_4 ▃▃▅▄▁▇▁▇▆▂▆▆▇▇▂▂▅▃▄█▆█▇▇▂▃▅▁▆▇▁▂▂▄▇▆▄▅▃▅
wandb: 
wandb: Run summary:
wandb: metric_0 0.53109
wandb: metric_1 0.00835
wandb: metric_2 0.64747
wandb: metric_3 0.76547
wandb: metric_4 0.64586
wandb: 
wandb: You can sync this run to the cloud by running:
wandb: wandb sync /Users/michaelvolk/Documents/projects/torchcell/wandb/offline-run-20240828_164829-jrz0tcmo
wandb: Find logs at: ./wandb/offline-run-20240828_164829-jrz0tcmo/logs
wandb: WARNING The new W&B backend becomes opt-out in version 0.18.0; try it out with `wandb.require("core")`! See https://wandb.me/wandb-core for more information.

--- Final Summary Contents ---

Contents of /Users/michaelvolk/Documents/projects/torchcell/wandb/run-20240828_164817-fktu3sxg/files/wandb-summary.json:
{"metric_0": 0.715624369777277, "metric_1": 0.2355671276911444, "metric_2": 0.6348888401726657, "metric_3": 0.7421840352522208, "metric_4": 0.42073917912566927, "_timestamp": 1724881704.603507, "_runtime": 7.228076934814453, "_step": 99, "_wandb": {"runtime": 7}}

Contents of /Users/michaelvolk/Documents/projects/torchcell/wandb/offline-run-20240828_164829-jrz0tcmo/files/wandb-summary.json:
{"_wandb": {"runtime": 6}}
```

This issue is important for users on compute nodes that either run many jobs in parallel and will therefore face rate limits logging to wandb or if their compute node has no internet so runs have to be synced after running jobs offline.

I was also able to reproduce these results on [NCSA Delta](https://docs.ncsa.illinois.edu/systems/delta/en/latest/#)

`Linux dt-login01.delta.ncsa.illinois.edu 4.18.0-477.58.1.el8_8.x86_64 #1 SMP Wed May 22 13:46:53 EDT 2024 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by kptkin** [comment]
> Hi there. I have an update on this with some minimal scripts to help reproduce the issue.
> 
> Using a conda env.
> 
> **Wand Version:** 0.17.7
> 
> **Chip**: Apple M1
> 
> **OS:** Sonoma 14.6.1
> 
> **Python Version:** 3.11.7
> 
> Here is the minimal script. It loops through some steps and logs data. The last of which we would expect to end up in the `wandb-summary.json`.
> 
> ```python
> #wandb_offline_debug.py
> import wandb
> import random
> import json
> import os
> import time
> 
> 
> def run_experiment(mode):
>     # Initialize wandb
>     run = wandb.init(project="wandb-offline-debug", mode=mode)
> 
>     # Log some random data
>     for step in range(100):
>         data = {f"metric_{i}": random.random() for i in range(5)}
>         wandb.log(data)
> 
>     # Print the summary
>     print(f"\nMode: {mode}")
>     print("Summary:")
>     summary_dict = wandb.run.summary._as_dict()
>     print(json.dumps(summary_dict, indent=2))
> 
>     wandb.finish()
>     return run.dir
> 
> 
> def print_summary_contents(run_dir):
>     summary_file_path = os.path.join(run_dir, "wandb-summary.json")
>     if os.path.exists(summary_file_path):
>         print(f"\nContents of {summary_file_path}:")
>         with open(summary_file_path, "r") as f:
>             print(f.read())
>     else:
>         print(f"\nSummary file not found at {summary_file_path}")
> 
> 
> if __name__ == "__main__":
>     # Run in online mode
>     online_run_dir = run_experiment("online")
> 
>     # Run in offline mode
>     offline_run_dir = run_experiment("offline")
> 
>     # Print contents of both summary files after runs finish
>     print("\n--- Final Summary Contents ---")
>     print_summary_contents(online_run_dir)
>     print_summary_contents(offline_run_dir)
> ```
> 
> From the results we see that std out shows the logged metrics and the summary in the `Run Summary`. Yet when we look at the `wandb-summary.json` these data only appears for the `online` run. When you go to sync the runs obviously the summary metrics won't appear for the offline runs. Just look under `--- Final Summary Contents ---`, the keys of json should match.
> 
> ```shell
> michaelvolk@M1-MV torchcell % /Users/michaelvolk/opt/miniconda3/envs/torchcell/bin/python /Users/michaelvolk/Documents/projects/torchce
> ll/wandb_offline_debug.py
> wandb: Currently logged in as: mjvolk3 (zhao-group). Use `wandb login --relogin` to force relogin
> wandb: Tracking run with wandb version 0.17.7
> wandb: Run data is saved locally in /Users/michaelvolk/Documents/projects/torchcell/wandb/run-20240828_164817-fktu3sxg
> wandb: Run `wandb offline` to turn off syncing.
> wandb: Syncing run stellar-puddle-5
> wandb: ⭐️ View project at https://wandb.ai/zhao-group/wandb-offline-debug
> wandb: 🚀 View run at https://wandb.ai/zhao-group/wandb-offline-debug/runs/fktu3sxg
> 
> Mode: online
> Summary:
> {
>   "metric_0": 0.715624369777277,
>   "metric_1": 0.2355671276911444,
>   "metric_2": 0.6348888401726657,
>   "metric_3": 0.7421840352522208,
>   "metric_4": 0.42073917912566927,
>   "_timestamp": 1724881704.603507,
>   "_runtime": 7.228076934814453,
>   "_step": 99
> }
> wandb:                                                                                
> wandb: 
> wandb: Run history:
> wandb: metric_0 ▅▄▂▇▆▄▅▃▄█▅▁▁▂▅▄██▇▅▇▆▄▂▃▆▄▃▁▁▂▄▄▆▇▆▁▃▆▆
> wandb: metric_1 ▇▂▃▅▂▆█▆▇▂▃▂▅▇█▇▃▅▅▃▃▇▆▃▆▃▇█▆▃▅▁█▂▇▃▄▅▇▂
> wandb: metric_2 █▁▃▄▁▄▄▆█▅▅▃▃▅▆▁▃▂▂▄▂█▃▇▆▄▃▇█▇▂▆▇▁█▃█▇▇▅
> wandb: metric_3 ▂▄▃▄▁▆▆▁▆▆▆▅▆▄█▆▇▄▅█▇█▃▃▆█▇▁▃▄▂▄█▄▃▁▇▆▁▆
> wandb: metric_4 ▄▄▁▁▁▇▆█▆▃▄▄▂▄▂▇▇▄█▃▄▄▂▂▃▇▃▅▄▄▂▅▆▂▂▆█▆▄▄
> wandb: 
> wandb: Run summary:
> wandb: metric_0 0.71562
> wandb: metric_1 0.23557
> wandb: metric_2 0.63489
> wandb: metric_3 0.74218
> wandb: metric_4 0.42074
> wandb: 
> wandb: 🚀 View run stellar-puddle-5 at: https://wandb.ai/zhao-group/wandb-offline-debug/runs/fktu3sxg
> wandb: ⭐️ View project at: https://wandb.ai/zhao-group/wandb-offline-debug
> wandb: Synced 6 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
> wandb: Find logs at: ./wandb/run-20240828_164817-fktu3sxg/logs
> wandb: WARNING The new W&B backend becomes opt-out in version 0.18.0; try it out with `wandb.require("core")`! See https://wandb.me/wandb-core for more information.
> wandb: Tracking run with wandb version 0.17.7
> wandb: W&B syncing is set to `offline` in this directory.  
> wandb: Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
> 
> Mode: offline
> Summary:
> {
>   "metric_0": 0.5310924540580804,
>   "metric_1": 0.008354595586691382,
>   "metric_2": 0.6474718406912846,
>   "metric_3": 0.7654659672380345,
>   "metric_4": 0.645856304637467,
>   "_timestamp": 1724881716.758178,
>   "_runtime": 7.470421075820923,
>   "_step": 99
> }
> wandb:                                                                                
> wandb: 
> wandb: Run history:
> wandb: metric_0 ▂▂▂▆▁▂▂▂▃▃▄▆▄▁▄▂▃▆▃▆▆▇▃▃▇▇▃▃▅▆▁▃▃▇█▅▄▃▃▅
> wandb: metric_1 ▆▂▅▆▂▃▃▃▇▄█▁▂▁▇▃▄▂▆▃▃▇▄▄▆▇▂██▃▃▃▄█▆▇▆▄▁▁
> wandb: metric_2 ▇▇▅▂▃▄█▄▅█▆█▄▂█▇▄▇▃▂▄▄▆▂▃▂▁▄▃▄▇▆▁▆▇▆▅▆█▆
> wandb: metric_3 ▂▇▂▅▂▄▂▆▁▇▃▇▃▇▁▂▄▂█▃▂▂▂▁▃▆▆▃▃▁█▅▆▁▄█▇▃▄▆
> wandb: metric_4 ▃▃▅▄▁▇▁▇▆▂▆▆▇▇▂▂▅▃▄█▆█▇▇▂▃▅▁▆▇▁▂▂▄▇▆▄▅▃▅
> wandb: 
> wandb: Run summary:
> wandb: metric_0 0.53109
> wandb: metric_1 0.00835
> wandb: metric_2 0.64747
> wandb: metric_3 0.76547
> wandb: metric_4 0.64586
> wandb: 
> wandb: You can sync this run to the cloud by running:
> wandb: wandb sync /Users/michaelvolk/Documents/projects/torchcell/wandb/offline-run-20240828_164829-jrz0tcmo
> wandb: Find logs at: ./wandb/offline-run-20240828_164829-jrz0tcmo/logs
> wandb: WARNING The new W&B backend becomes opt-out in version 0.18.0; try it out with `wandb.require("core")`! See https://wandb.me/wandb-core for more information.
> 
> --- Final Summary Contents ---
> 
> Contents of /Users/michaelvolk/Documents/projects/torchcell/wandb/run-20240828_164817-fktu3sxg/files/wandb-summary.json:
> {"metric_0": 0.715624369777277, "metric_1": 0.2355671276911444, "metric_2": 0.6348888401726657, "metric_3": 0.7421840352522208, "metric_4": 0.42073917912566927, "_timestamp": 1724881704.603507, "_runtime": 7.228076934814453, "_step": 99, "_wandb": {"runtime": 7}}
> 
> Contents of /Users/michaelvolk/Documents/projects/torchcell/wandb/offline-run-20240828_164829-jrz0tcmo/files/wandb-summary.json:
> {"_wandb": {"runtime": 6}}
> ```
> 
> This issue is important for users on compute nodes that either run many jobs in parallel and will therefore face rate limits logging to wandb or if their compute node has no internet so runs have to be synced after running jobs offline.
> 
> I was also able to reproduce these results on [NCSA Delta](https://docs.ncsa.illinois.edu/systems/delta/en/latest/#)
> 
> `Linux dt-login01.delta.ncsa.illinois.edu 4.18.0-477.58.1.el8_8.x86_64 #1 SMP Wed May 22 13:46:53 EDT 2024 x86_64 x86_64 x86_64 GNU/Linux`
> 
> edit:
> 
> I did a little more debugging and if you inspect that state in this method ([_update_summary](https://github.com/wandb/wandb/blob/d811a2e19b20cecc311973f7874a9c0f88e09c9b/wandb/sdk/internal/sender.py#L1226C39-L1240C68)) you can see that for `offline` it doesn't update properly.

Just to clarify the data is save locally, it is save to our transaction log (.wandb file in your run folder), we indeed don't populate the summary file, although it will be populated when you do sync the run to the wandb backend (you can verify it by running the `wandb sync` command on your run folder)

To give a bit more context: this file is mainly used for backup purposes, as we actually send the summary values using a different interface.

@Mjvolk3 could you please clarify your ask here? thanks

**Comment by Mjvolk3** [comment]
@kptkin I just tried to run `wandb sync` and it worked showing the value properly populated in the online interface. Previously, when I ran sync on offline runs the values were not being populate online. This was occurring in a different more complicated script not shown here, so maybe the issue is somewhere within my own code (likely runs not completing properly so there is no logging). If the issue shows again and I think it is related to `wandb sync` I will comment back. Thank you for clarifying the use of different interfaces in `online` vs `offline`. We can keep the issue closed. Thanks 😊

**Comment by kptkin** [comment]
> @kptkin I just tried to run `wandb sync` and it worked showing the value properly populated in the online interface. Previously, when I ran sync on offline runs the values were not being populate online. This was occurring in a different more complicated script not shown here, so maybe the issue is somewhere within my own code (likely runs not completing properly so there is no logging). If the issue shows again and I think it is related to `wandb sync` I will comment back. Thank you for clarifying the use of different interfaces in `online` vs `offline`. We can keep the issue closed. Thanks 😊

Thanks for the clarification! Please don't hesitate to open another issue for us if you encounter some problem.

**Comment by CedricLeon** [comment]
I'm just seeing that now, many thanks @Mjvolk3 for the minimal reproduction and @kptkin for the answer. This confirms me that my problem (in my more complicated script) was linked to the runs not completing correctly.
I did not encounter the issue again since I updated my W&B `init` and `finish` functions. I did not modified anything obvious or major though, so my wild guess is that a third party programs (I mentioned Hydra and Submitit) interferes with the process.
